### DISCUSSION TITLE
Adventurer 5M Klipper Mod v00.05-beta

### DISCUSSION THREAD

**Discussion by xblax** [xblax/flashforge_ad5m_klipper_mod#173] [Announcements]
Unoffical mod for Flashforge Adventurer 5M (Pro) 3D printers to run Moonraker, custom Klipper, Mainsail & Fluidd. 
The Klipper Mod for ADM5 (Pro) is currently provided in two variants:

- **"Lite"** variant without on-screen GUI control application
- **"KlipperScreen"** variant. 
If you encounter any issues that could be caused by resource exhaustion (mostly system RAM), please check if the issue also occurs with the lite variant.  If you experience wifi issues or have no wifi stations when using Klipperscreen, please save the `/var/log/KlipperScreen.log` file and post it as an issue if possible.
- **"GuppyScreen"** variant. *Currently included as a preview in alpha state.* It offers a more lightweight UI on the limited hardware.

Before installation read through the [Readme](https://github.com/xblax/flashforge_adm5_klipper_mod/blob/master/README.md) and other provided documentation. This release is known to work with Flashforge firmware versions 2.4.5, 2.5.5 and 2.5.6.

### Changes

* enable support for Flashforge firmware versions > 2.4.5
* guppyscreen: add new build variant with alpha preview of [Guppy Screen](https://github.com/ballaswag/guppyscreen) UI (see discussion https://github.com/xblax/flashforge_ad5m_klipper_mod/discussions/134)
* moonraker-timelapse: integrate plugin, optimized ffmpeg settings for low memory usage (see issue https://github.com/xblax/flashforge_ad5m_klipper_mod/issues/119)
* klipper: add G17 support for spiral Z-Hop (see issue https://github.com/xblax/flashforge_ad5m_klipper_mod/issues/48)
* klipper: add `AUTO_SHUTDOWN` macro (see issue https://github.com/xblax/flashforge_ad5m_klipper_mod/issues/56 as suggested by @Tiwatz)
* klipper: fix servo angle bug (mentioned in issue https://github.com/xblax/flashforge_ad5m_klipper_mod/issues/38 by @Tiwatz)
* klipper: enable support for firmware retractions
* klipper: add support for optional skew correction profile
* klipperscreen: custom g2d accelerated fbturbo driver (see issue https://github.com/xblax/flashforge_ad5m_klipper_mod/issues/18)
* klipperscreen: clean-up of buildroot dependencies to reduce size / memory usage (see issue https://github.com/xblax/flashforge_ad5m_klipper_mod/issues/18)
* mainsail: update to version 2.11.1
* fluidd: update to version 1.30.0
* buildroot: update to 2023.02.11 and improve structure of mod specific customizations (see issue https://github.com/xblax/flashforge_ad5m_klipper_mod/issues/32, ongoing work)
* other minor fixed and optimizations (see git log)

### Known issues
Please check the [Issues](https://github.com/xblax/flashforge_ad5m_klipper_mod/issues) section of the GitHub repository.

---
> [!WARNING]
> *Make sure to backup your data and changed config files before [updating](https://github.com/xblax/flashforge_adm5_klipper_mod/blob/master/docs/INSTALL.md#updating)!*
>
> Clear web browser cache after installing the update if Mainsail or Fluidd update causes any issues. 

> [!CAUTION]
> *If you want to install this mod to your ADM5 (Pro) then be aware, that you risk to loose your warranty or damage the printer. Proceed at your own risk if you want to try this mod!*

<hr /><em>This discussion was created from the release <a href='https://github.com/xblax/flashforge_ad5m_klipper_mod/releases/tag/v00.05-beta'>Adventurer 5M Klipper Mod v00.05-beta</a>.</em>

**Comment by KaruroChori** [comment]
I just tested the guppyscreen version, and it seems to be working well. Many thanks to @ballaswag & @consp!

**Reply by MACCAMENZIES** [reply]
Do have issue with the accelerometer test. Seems it goes through the vibrations and then sits there with the two wheels going around . Guppyscreen

**Reply by MrSolidus89** [reply]
> Do have issue with the accelerometer test. Seems it goes through the vibrations and then sits there with the two wheels going around . Guppyscreen

but is it not possible to change the language?

**Reply by consp** [reply]
> but is it not possible to change the language?

Not at the moment,  this is a feature lacking from guppyscreen at this point afaik. Guppy is a bit less mature than klipperscreen but a lot lighter on the system so some features might be lacking at this time.

**Reply by consp** [reply]
> Do have issue with the accelerometer test. Seems it goes through the vibrations and then sits there with the two wheels going around . Guppyscreen

This is an unfortunate issue with numpy eating all resources and then failing. It happens quite often, I've been working on fixing it but so far it's hit-or-miss and not ready for release yet. Might be a while before it's fixed. See #175 for updates in the future

**Reply by ballaswag** [reply]
Guppyscreen needs matplotlib to plot those shaper graphs. I think it might import it even if the graph option is unchecked.

**Reply by xblax** [reply]
Maplotlib is quite heavy in size, I think it's like 20 or 30MB extra in the compressed chroot. Maybe it could be stripped down. Not sure how much RAM it needs. My reasoning was that plot can be generated externally on a separate PC if needed. But we might it include it if it's a feature of Guppyscreen.

**Reply by consp** [reply]
If I remember (from a vague blue moon time when I used it somewhere, memory is fuzzy about it though I could be wrong), it is so big due to the numpy dependency. But we already need that for klipper.

**Comment by nephtyss** [comment]
Can anyone confirm this release works on firmware 2.6.6?

**Reply by consp** [reply]
It only checks for minversion 2.4.5 now so it should install. ~~Have not checked the install files yet so ymmv, do you know if the firmware is downloadable somewhere or only OTA?~~

Never mind, this was already released, it only updates the flashforge gui software (which controls everything in the normal firmware), no kernel updates or other changes so it should work without issues.

**Reply by consp** [reply]
On another note: flashforge updated their site and removed the old firmware. There is now no way to officially get the firmware install image. I guess they really don't want users installing via usb.

**Reply by xblax** [reply]
It seems they still have firmware available, but not the new versions and not in ".tgz"-format:
https://flashforge.com/blogs/download-document/guider-3

Looks like they are re-modeling their website. Also weird they name the printer "Guider 3" in the URL.

**Reply by consp** [reply]
Guider 3 is also in some of the files on the original firmware iirc, maybe internal name.

That page went 404 when I looked, they updated it afterwards I guess

**Reply by RyantHults** [reply]
has anyone been able to get this running on an A5M on 2.6.6? I loaded the .tgz on a flashdrive, stuck it into the printer (usb port on the screen) and it booted up normally, no update screen.

**Reply by consp** [reply]
Haven't tested it yet, if I do not forget I'll try it tonight.

Obligatory questions:
* Did you download the correct one? As in pro vs non pro?
* Is the drive FAT32 formatted?

**Reply by RyantHults** [reply]
turns out it had been formatted as exfat, not fat32 (curse you apple!). formatted it as fat32 and it installed fine now.

**Reply by MACCAMENZIES** [reply]
Has anyone looked at changing the screen on the pro? Not sure if bigscreen would fit like big tree or similar.  Just curious

**Reply by IFVal** [reply]
@consp Firmware 2.7.1 and 2.7.2 sent by Flashforge support service on request. Use it if necessary.
https://drive.google.com/drive/folders/1xFtBskBtqXpetlgDQS_6A-9nmxBmWJ3B?usp=sharing

**Comment by johnnyutah775** [comment]
what file do we rename for the pro. im a noob. but i love how my qidi runs on klipper i downloaded the release inside it says chroot.tar and is a zip img folder and doc that says flashforge_init.sh

**Reply by RyantHults** [reply]
you should not unzip/untar it. you just put the Adventurer5MPro-KlipperMod-v00.05-*.tgz file on a FAT32 formatted flashdrive and follow the install steps

**Reply by johnnyutah775** [reply]
its not working for me maybe the firm where is 2.6.6?

**Reply by johnnyutah775** [reply]
i also have a pro.

**Reply by johnnyutah775** [reply]
> you should not unzip/untar it. you just put the Adventurer5MPro-KlipperMod-v00.05-*.tgz file on a FAT32 formatted flashdrive and follow the install steps

it says rename archive files i have zero idea what that means

**Reply by xblax** [reply]
> it says rename archive files i have zero idea what that means

Yes, that's outdated info, sorry - we have to update the docs. No renaming is needed. We do now publish both non-pro and pro variants of the update files. Previously the non-pro variant had be renamed for the pro version of the printer.

**Reply by RyantHults** [reply]
> its not working for me maybe the firm where is 2.6.6?

make sure the flashdrive is FAT32. Mine did not work at first until I figured out it had been formatted at exFAT instead of FAT32

**Comment by cwestpha** [comment]
2.6.7 and tried this. Klipper + Moonraker + Fluidd + guppy seems to work. Haven't tested mainsail. Would be nice if you could remove the web UX you don't use to free up a little more system ram.
some space for macro improvement but no surprise there since its a .5 release so there is always fine tuning.
Main issues seem related to possible firewall now blocking some ports on newer firmware. Cant SSH (no route over port 22) cant upload files directly from Orca (I am betting again ports blocking). _Another issue is webcam isnt working with no log of the camera server trying to init. Since there is no SSH I cant confirm, but I am pretty sure something is blocking that service from even starting or being registered._
Update: Turns out it doesnt write a log or anything. odd.
http://[printer'sIP/FQDN]:8080?action=stream
http://[printer'sIP/FQDN]:8080?action=snapshot

**Reply by GAMRABADR** [reply]
accelerometro e bed level sono sballati ?

**Reply by KaruroChori** [reply]
@GAMRABADR, I am just going to say this once: this repository is using English as its language of reference, so that everyone can understand each other.  
Not only that, but your previous messages were angry, impolite and filled to the brim with errors which would have prevented any translation of them (which is why I removed them). You can write in whatever language you prefer, but at the very least you should always attach a machine translation, and accept that is how most of us are going to read them.
Finally, if you have specific issues this is not the best place. Open an `issue` for anything you think might be a new bug or a new conversation to discuss the installation process if you are not familiar with it, but don't hijack a general thread like this one.

**Comment by staniox** [comment]
when i put a file named klipper_mod_skip on a USB drive and plug it in before printer start it doesn't work any solution?

**Comment by vennen894** [comment]
Any plans for further development? I have been running for about a week now and absolutely love what you have done here!!!

Thank you for allowing me to use a product that I paid for (like it should be)!!

**Reply by AJolly** [reply]
Its still in development, but we are hampered by flashforge not releasing full sources for klipper

**Comment by ShayGus** [comment]
Will it work on 2.7.5?

**Reply by vennen894** [reply]
I am running 2.7.5 and have no issues switching back and forth.

I love this mod and have spent a lot of time to customize the macros to my personal liking.

**Reply by cwestpha** [reply]
> I am running 2.7.5 and have no issues switching back and forth.
> 
> I love this mod and have spent a lot of time to customize the macros to my personal liking.

have you gotten SSH to work with that version?
So far I got everything but that to work.

**Reply by vennen894** [reply]
SSH while the mod is running yes. Using MobaXterm.

Make sure your using root and password of klipper

**Reply by zorn2u** [reply]
perfect, good to know! i did wait for that before i order:-)
do you use guppy? read there a issues with inputshaper. will that make problems in daily use?

**Reply by xblax** [reply]
Got do know it's working. I didn't switch to stock firmware and update it in quite some time ...

Didn't have much time to contribute to anything in the last months. Maybe in the next weeks we can see if there's something worth making maintenance release.

**Comment by cyborg-sova** [comment]
When I enter my WiFi password and click "Connect", the button is pressed, but the connection doesn't seem to happen. Is this what the author was referring to? v2.7.6

**Comment by kr0sev** [comment]
Gonna start writing up the comment because I feel like I'm going crazy here!

Attempting to install guppyscreen on AD5M (not pro) 2.7.9, formatted the drive to FAT32, tried both zipped and unzipped, and no matter what I do it's just booting up normally. 

When I _did_ format it was a quick format, so going back right now and doing a full format to try it all again. If it works I'll drop a comment but on the chance it doesn't I'm looking for any assist!

**Reply by kr0sev** [reply]
UPDATE: Doing a full format instead of a quick format fixed the issue! (also changed the data allocation size to system default, can't remember what it was set to before) Hopefully this helps anyone that gets stuck where I was!

**Comment by Simmondsp** [comment]
I have installed klipper screen on the AD5MP. but it loads up as the Guppy Screen? not sure why or what is needed to change this?

**Reply by MACCAMENZIES** [reply]
Wrong firmware. Download the other one (klipperscreen)

**Reply by Cilo456** [reply]
Guppy screen is better anyways

**Comment by zorn2u** [comment]
anyone installed on newest fw 3.1.3? since orcaflashforge works smoothly, what are the major advantages? not sure what i am missing ; - )

**Reply by ffontanac** [reply]
No more MCU errors since updated last week (crossed fingers...). Udated to 3.1.4

**Reply by killernos** [reply]
i am running 3.1.5 with klipper it seem buggy none of the extrude commands work right on the machine, i have to keep doing everything from computer, i just got this 3d printer.

**Comment by DragonMaster518** [comment]
I need help downloading the files

**Comment by Travis3895** [comment]
My AD5M will not print. Everything else works, I can home and extrude filament, but it keeps telling me there is no filament detected in the runout sensor. I have plenty of filament loaded on the printer, it started to print today but quickly paused and told me to load filament. How do I fix this so that my printer will print again? I am running Kilpper Mod v00.05-beta (I double checked in case that was the issue) and I am using guppy screen.

**Reply by MACCAMENZIES** [reply]
Did you try turning off the filament sensor . Known issue with the detection if you read the pull requests

**Reply by Travis3895** [reply]
I disabled the filament sensor and it still gives errors, I watched the console this time and it heats the bed up and then tells me I have to home the printer first. So I homed all axes and tried to reprint, the bed moved up then threw an error "Extrude below minimum temp" what am I doing wrong? It hasn't printed since I installed the firmware

**Reply by MACCAMENZIES** [reply]
Not familiar with the guppy screen as I run the full display. I would have said try and heat the nozzle from the machine screen. Try and manually heat the nozzle from fluidd?

**Reply by Travis3895** [reply]
The UI I chose shouldn't affect much in comparison to the full display you use with the exception of maybe I don't have as many options. But I'm starting the print and watching the console in fluidd. I'm just not sure why when I start a print it's heating up the bed but not the extruder and throwing an error that the extruder isn't hot enough.

**Reply by Tiwatz** [reply]
Probably something wrong with your start g-code.

**Reply by MACCAMENZIES** [reply]
> The UI I chose shouldn't affect much in comparison to the full display you use with the exception of maybe I don't have as many options. But I'm starting the print and watching the console in fluidd. I'm just not sure why when I start a print it's heating up the bed but not the extruder and throwing an error that the extruder isn't hot enough.

I'm suggesting to manually heat your extruder from either the screen or from fluidd to see if it's a fault with your extruder or it's your gcode

**Comment by k7bfkjpdqt-create** [comment]
I'm not getting on with this mod. Trying to uninstall and every time it says Klipper mod uninstalled please reboot and upon reboot it boots back up with the Klipper mod still installed. Anyone any ideas it's rather infuriating
